mlflow/server/auth/__init__.py

```diff
@@ -485,15 +485,23 @@ def set_can_manage_experiment_permission(resp: Response):
     parse_dict(resp.json, response_message)
     experiment_id = response_message.experiment_id
     username = authenticate_request().username
-    store.create_experiment_permission(experiment_id, username, MANAGE.name)
+    permission = _get_permission_from_store_or_default(
+        lambda: store.get_experiment_permission(experiment_id, username).permission
+    )
+    if permission != MANAGE.name:
+        store.create_experiment_permission(experiment_id, username, MANAGE.name)
```

Comment:
For `MultipleResultsFound` cases, should we remove the existing permissions for (experiment_id, name) first then add the new one to avoid such errors? @gabrielfu